tests/cp.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tests
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	"github.com/runfinch/common-tests/command"
+	"github.com/runfinch/common-tests/ffs"
+	"github.com/runfinch/common-tests/option"
+)
+
+// Cp tests `finch cp` command to copy files between container and host filesystems.
+func Cp(o *option.Option) {
+	filename := "test-file"
+	content := "test-content"
+	containerFilepath := filepath.Join("/tmp", filename)
+	containerResource := fmt.Sprintf("%s:%s", testContainerName, containerFilepath)
+
+	ginkgo.Describe("copy from container to host and vice versa", func() {
+		ginkgo.BeforeEach(func() {
+			command.RemoveAll(o)
+		})
+		ginkgo.AfterEach(func() {
+			command.RemoveAll(o)
+		})
+
+		ginkgo.Context("when the container is running", func() {
+			ginkgo.BeforeEach(func() {
+				command.Run(o, "run", "-d", "--name", testContainerName, defaultImage, "sleep", "infinity")
+			})
+
+			ginkgo.It("should be able to copy file from host to container", func() {
+				path := ffs.CreateTempFile(filename, content)
+				ginkgo.DeferCleanup(os.RemoveAll, filepath.Dir(path))
+
+				command.Run(o, "cp", path, containerResource)
+				fileShouldExistInContainer(o, testContainerName, containerFilepath, content)
+			})
+
+			ginkgo.It("should be able to copy file from container to host", func() {
+				cmd := fmt.Sprintf("echo -n %s > %s", content, containerFilepath)
+				command.Run(o, "exec", testContainerName, "sh", "-c", cmd)
+				fileDir := ffs.CreateTempDir("finch-test")
+				path := filepath.Join(fileDir, filename)
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				command.Run(o, "cp", containerResource, path)
+				fileShouldExist(path, content)
+			})
+
+			for _, link := range []string{"-L", "--follow-link"} {
+				ginkgo.It(fmt.Sprintf("with %s flag, should be able to copy file from host to container and follow symbolic link",
```

Comment:
The variable needs to be redefined here: more info - https://github.com/runfinch/tests/pull/15#discussion_r984021849
```suggestion
			for _, link := range []string{"-L", "--follow-link"} {
			         link := link
				ginkgo.It(fmt.Sprintf("with %s flag, should be able to copy file from host to container and follow symbolic link",
```